src/main/java/org/commcare/session/RemoteQuerySessionManager.java

```diff
@@ -143,13 +143,17 @@ public Multimap<String, String> getRawQueryParams(boolean skipDefaultPromptValue
     }
 
     private EvaluationContext getEvaluationContextWithUserInputInstance() {
-        Map<String, String> userQueryValues = getUserQueryValues(false);
-        ExternalDataInstance userInputInstance = VirtualInstances.buildSearchInputInstance(userQueryValues);
+        ExternalDataInstance userInputInstance = getUserInputInstance();
         return evaluationContext.spawnWithCleanLifecycle(
                 ImmutableMap.of(userInputInstance.getInstanceId(), userInputInstance)
         );
     }
 
+    public ExternalDataInstance getUserInputInstance() {
+        Map<String, String> userQueryValues = getUserQueryValues(false);
+        return VirtualInstances.buildSearchInputInstance(userQueryValues);
```

Comment:
TODO: the instance name needs to be unique for the session - one option could be to use `query/@storage-instance` with a prefix e.g. `resutls-search-input`.

On the topic of naming, is `search-input` right? What about `user-input`, `query-prompts`, `query-data`?

Reply:
Also see the discussion here: https://github.com/dimagi/commcare-core/pull/1082#issuecomment-1134448655